### ISSUE TITLE
Construct plugin main class in onEnable

### ISSUE THREAD

**Issue by SOF3** [pmmp/PocketMine-MP#3067]
## Conditions
This issue depends on #2825 that allows removing `onLoad`.

## Motivation
If the main class of plugin "Foo" uses some static code from another plugin "Bar" in onEnable, and "Foo" is in phar format, and "Bar" is in folder format to be loaded by DevTools, "Foo" has to soft-depend on DevTools so that it gets loaded after "Bar" is in the class path.
(Truth: This is probably another bug: Perhaps this simply won't work because "Foo" can't load due to missing dependency; not tested to find out which is the case)

This issue was previously encountered by @thebigsmileXD in DEVirion, and I had to move the main code of DEVirion to `onLoad`, which is a bad idea, and the other plugin had to soft-depend on DEVirion.

However, the above can be mitigated with improved plugin dependency management using the load--SCAN--sort mechanism as introduced by #3064.

This issue is a bigger problem when the plugin implements some interface from a dependency plugin (or extends some class, but this is most likely a bad idea). As the dependency class is required when the main class is loaded, especially when `onLoad` is to be removed.

## Proposed changes
The issue can be mitigated by constructing the main class just before `onEnable` is called.

**Comment by dktapps** [comment]
The DevTools issue was fixed by https://github.com/pmmp/PocketMine-MP/pull/3971.

**Comment by dktapps** [comment]
Closing in favour of #6330
